src/plugins/orangehrmPerformancePlugin/Api/EmployeeTrackerAPI.php

```diff
@@ -0,0 +1,203 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Performance\Api;
+
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\CollectionEndpoint;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointCollectionResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Traits\Auth\AuthUserTrait;
+use OrangeHRM\Performance\Api\Model\EmployeeTrackerModel;
+use OrangeHRM\Performance\Dto\EmployeeTrackerSearchFilterParams;
+use OrangeHRM\Performance\Service\EmployeeTrackerService;
+
+class EmployeeTrackerAPI extends Endpoint implements CollectionEndpoint
+{
+    use AuthUserTrait;
+
+    public const FILTER_INCLUDE_EMPLOYEES = 'includeEmployees';
+    public const FILTER_NAME_OR_ID = 'nameOrId';
+
+    public const PARAM_RULE_FILTER_NAME_OR_ID_MAX_LENGTH = 100;
+
+    private const ADMIN_USER_ROLE_ID = 1;
```

Comment:
don't hard code user role ids or user role names. need to handle through user role manager